caddyhttp/httpserver/server.go

```diff
@@ -160,6 +160,18 @@ func (s *Server) Serve(ln net.Listener) error {
 		ln = tcpKeepAliveListener{TCPListener: tcpLn}
 	}
 
+	if ProxyProtocolSubnets != "" {
+		// We decode proxy protocol after the TCP wrapper, but before
+		// anything else. As long as the wrapper happens before things
+		// that care about RemoteAddr, we should be ok.
+		fln, err := newFilteredPROXYListener(ln, ProxyProtocolTimeout, ProxyProtocolSubnets)
+		if err != nil {
+			fmt.Printf("[ERROR] configuring Proxy Protocol support: %v\n", err)
```

Comment:
I agree. I actually had an issue today where kubernetes started a pod (using a different plugin) that didn't initialize properly, and it took some wasted time to see that it shouldn't have started in the first place. Lesson learned, and I think it applies here.